addons/project/controllers/portal.py

```diff
@@ -543,3 +545,43 @@ def portal_my_task(self, task_id, report_type=None, access_token=None, project_s
             request.session['my_tasks_history'] = task_sudo.ids
         values = self._task_get_page_view_values(task_sudo, access_token, **kw)
         return request.render("project.portal_my_task", values)
+
+    @http.route('/project_sharing/attachment/add_image', type='http', auth='user', methods=['POST'], website=True)
+    def add_image(self, name, data, res_id, access_token=None, **kwargs):
+        res_model = 'project.task'
+        try:
+            task_sudo = self._document_check_access(res_model, int(res_id), access_token=access_token)
```

Comment:
```suggestion
        try:
            task_sudo = self._document_check_access('project.task', int(res_id), access_token=access_token)
```
clearer and be sure to catch it during review if we change our mind later